src/unity/toolkits/object_detection/one_shot_object_detection/util/superposition.cpp

```diff
@@ -3,119 +3,99 @@
+  // background from RGB to RGBA
+  boost::gil::copy_and_convert_pixels(
+    starter_image_view,
+    view(starter_image_rgba)
+  );
+  resample_pixels(view(starter_image_rgba), transformed, M, boost::gil::bilinear_sampler());
+  color_quadrilateral(transformed, parameter_sampler.get_warped_corners());
+  superimpose_image(masked, transformed, background);
 }
 
-flex_image create_synthetic_image_from_rgba_object(const boost::gil::rgb8_image_t::view_t &background_view, 
-                                       ParameterSampler &parameter_sampler,
-                                       const flex_image &object) {
+void superimpose_rgba_object_image(ParameterSampler &parameter_sampler,
+                                   const flex_image &object,
+                                   const boost::gil::rgb8_image_t::view_t &masked,
+                                   const boost::gil::rgba8_image_t::view_t &transformed,
+                                   const boost::gil::rgba8_image_t::view_t &background) {
   Eigen::Matrix<float, 3, 3> M = parameter_sampler.get_transform().inverse();
   boost::gil::rgba8_image_t::view_t starter_image_view = interleaved_view(
     object.m_width,
     object.m_height,
     (boost::gil::rgba8_pixel_t*) (object.get_image_data()),
     object.m_channels * object.m_width // row length in bytes
   );
+  resample_pixels(starter_image_view, transformed, M, boost::gil::bilinear_sampler());
+  superimpose_image(masked, transformed, background);
+}
+
+flex_image create_synthetic_image(const boost::gil::rgb8_image_t::view_t &background_view,
+                                  ParameterSampler &parameter_sampler,
+                                  const flex_image &object) {
   boost::gil::rgba8_image_t background_rgba(boost::gil::rgba8_image_t::point_t(background_view.dimensions()));
-  // The following line uses the color_convert conversion to convert the 
-  // background from RGB to RGBA
   boost::gil::copy_and_convert_pixels(
     background_view,
-    boost::gil::view(background_rgba)
+    view(background_rgba)
   );
   boost::gil::rgba8_image_t transformed(boost::gil::rgba8_image_t::point_t(background_view.dimensions()));
   fill_pixels(view(transformed), RGBA_WHITE);
-  resample_pixels(starter_image_view, view(transformed), M, boost::gil::bilinear_sampler());
-  boost::gil::rgb8_image_t masked(boost::gil::rgb8_image_t::point_t(background_view.dimensions()));
-  fill_pixels(view(masked), WHITE);
-  superimpose_image_rgba(view(masked), view(transformed), 
-                         view(background_rgba));
+  boost::gil::rgb8_image_t masked(boost::gil::rgba8_image_t::point_t(background_view.dimensions()));
+  fill_pixels(view(masked), RGB_WHITE);
+  if (object.m_channels == 4) {
+    superimpose_rgba_object_image(parameter_sampler, object, view(masked), view(transformed), view(background_rgba));
+  } else {
+    superimpose_rgb_object_image(parameter_sampler, object, view(masked), view(transformed), view(background_rgba));
```

Comment:
Could we combine `superimpose_rgb_object_image` and `superimpose_rgba_object_image` into one function that introspects the format of the `flex_image` internally?

Reply:
That was my thought behind `create_synthetic_image` which is the only function exposed in `superposition.hpp`. `superimpose_rgb_object_image` and `superimpose_rgba_object_image` are meant to be helper functions for `create_synthetic_image`.

Reply:
The two functions seem to share much of their logic. I guess what I don't understand is why the RGB case (for the user's object image) can't be handled just by converting from RGB to RGBA at the very beginning, and then having everything else be the same code path.

Boost GIL might even have some way of producing an RGBA view over an RGB image

Reply:
Fixed.